File: cpp/easy/search_insert_position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if(nums.size() == 0)
            return 0;
        int begin = 0;
        int end = nums.size()-1;
        while(begin <= end)
        {
            int mid = begin + (end-begin)/2;
            if(nums[mid] == target)
                return mid;
            else if(nums[mid] < target)
                begin = mid+1;
            else
                end = mid-1;
        }
        return begin;
    }
};

Conclusion:
Easy issue, binary search, time complexity is: O(logN), N is the number of elements in the input array. Space Complexity is: O(1)

File: python/easy/search_insert_position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        if len(nums) == 0:
            return 0
        start, end = 0, len(nums)-1
        while start <= end:
            mid = start + (end - start)/2
            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                start = mid + 1
                if start >= len(nums):
                    return len(nums)
                else:
                    if nums[start] > target:
                        return start
            else:
                end = mid - 1
                if end < 0:
                    return 0
                else:
                    if nums[end] < target:
                        return end+1
